src/Components/Components/src/PersistentComponentState.cs

```diff
@@ -155,6 +161,82 @@ internal bool TryTakeFromJson(string key, [DynamicallyAccessedMembers(JsonSerial
         }
     }
 
+    /// <summary>
+    /// Registers a callback to be invoked when state is restored for a specific scenario.
+    /// If state for the scenario is already available, the callback is invoked immediately.
+    /// </summary>
+    /// <param name="scenario">The scenario for which to register the callback.</param>
+    /// <param name="callback">The callback to invoke during restoration.</param>
+    /// <returns>A subscription that can be disposed to unregister the callback.</returns>
+    public RestoringComponentStateSubscription RegisterOnRestoring(
+        IPersistentComponentStateScenario scenario, 
+        Action callback)
+    {
+        ArgumentNullException.ThrowIfNull(scenario);
+        ArgumentNullException.ThrowIfNull(callback);
+
+        var isRecurring = scenario.IsRecurring;
+        _restoringCallbacks.Add((scenario, callback, isRecurring));
```

Comment:
Why do you need to store IsRecurring, you already have it available within the scenario. you can just check scenario.IsRecurring